packages/transformer/src/test/standalone/IModelTransformerHub.test.ts

```diff
@@ -797,6 +794,71 @@ describe("IModelTransformerHub", () => {
     sinon.restore();
   });
 
+  it("should update aspects when processing changes and detachedAspectProcessing is turned on", async () => {
+    let elementIds: Id64String[] = [];
+    const aspectIds: Id64String[] = [];
+    const sourceIModelId = await createPopulatedIModelHubIModel("TransformerSource", async (sourceSeedDb) => {
+      elementIds = [
+        Subject.insert(sourceSeedDb, IModel.rootSubjectId, "Subject1"),
+        Subject.insert(sourceSeedDb, IModel.rootSubjectId, "Subject2"),
+      ];
+
+      // 10 aspects in total (5 per element)
+      elementIds.forEach((element) => {
+        for (let i = 0; i < 5; ++i) {
+          const aspectProps: ExternalSourceAspectProps = {
+            classFullName: ExternalSourceAspect.classFullName,
+            element: new ElementOwnsExternalSourceAspects(element),
+            identifier: `${i}`,
+            kind: "Document",
+            scope: { id: IModel.rootSubjectId, relClassName: "BisCore:ElementScopesExternalSourceIdentifier" },
+          };
+
+          const aspectId = sourceSeedDb.elements.insertAspect(aspectProps);
+          aspectIds.push(aspectId); // saving for later deletion
+        }
+      });
+
+      return Promise.resolve();
```

Comment:
why return `Promise.resolve()` explicitly in an async function? You don't even use `await`, so you could just make the caller allow non-promise results, and make this function not-async

Reply:
Removed `Promise.resolve()` and refactored method to not force async version.